File: core/src/main/java/com/web3auth/core/types/SHA512.java
package com.web3auth.core.types;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class SHA512 {
    public static byte[] digest(byte[] buf) throws NoSuchAlgorithmException {
        MessageDigest digest = MessageDigest.getInstance("SHA-512");
        digest.update(buf);
        return digest.digest();
    }
}

File: core/src/main/java/com/web3auth/core/types/SHA512.kt
package com.web3auth.core.types

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException


object SHA512 {
    @Throws(NoSuchAlgorithmException::class)
    fun digest(buf: ByteArray?): ByteArray {
        val digest = MessageDigest.getInstance("SHA-512")
        digest.update(buf)
        return digest.digest()
    }
}